core/src/main/scala/io/qbeast/core/model/AutoIndexer.scala

```diff
@@ -0,0 +1,36 @@
+package io.qbeast.core.model
+
+/**
+ * AutoIndexer interface to automatically choose which columns to index.
+ * @tparam DATA
+ *   the data to index
+ */
+trait AutoIndexer[DATA] {
+
+  /**
+   * The maximum number of columns to index.
+   * @return
+   */
+  def MAX_COLUMNS_TO_INDEX: Int
+
+  /**
+   * Chooses the columns to index without limiting the number of columns.
+   * @param data
+   *   the data to index
+   * @return
+   */
+  def chooseColumnsToIndex(data: DATA): Seq[String]
```

Comment:
Since MAX_COLUMNS_TO_INDEX is a part of the API, then this method can have a default implementation. It will reduce the work for those who is going to implement the trait.